handbook/index.md

```diff
@@ -27,6 +27,7 @@ The Sourcegraph handbook describes how we (Sourcegraph teammates) work. It's pub
 ## Engineering
 
 - [Engineering](engineering/index.md)
+- [Documentation](documentation/index.md)
```

Comment:
Why is this under engineering and not a new top level item (as the directory would suggest)? There is a bit of ambiguity here whether that page is product documentation or handbook documentation.

```suggestion

## Documentation

- [Documentation](documentation/index.md)
```

Reply:
> There is a bit of ambiguity here whether that page is product documentation or handbook documentation

@nicksnyder I agree and here are some additional changes I've made to help address this:

- Removed the documentation link on the home page
- I've pushed a new change that moves the documentation page under engineering
- I've added a paragraph to that documentation page explaining its documentation for the sourcegraph repo with a link to jump to how to edit the handbook
- Added a link to editing the handbook on the handbook home page